packages/runtime-core/src/apiSetupHelpers.ts

```diff
@@ -268,6 +317,41 @@ export function useAttrs(): SetupContext['attrs'] {
   return getContext().attrs
 }
 
+export function useModel<T>(name: string): WritableComputedRef<T> {
+  const i = getCurrentInstance()!
+  if (__DEV__ && !i) {
+    warn(`useModel() called without active instance.`)
+    return null as any
+  }
+
+  const options = (i.propsOptions[0] as NormalizedProps)[name]
+  if (options && !options.required) {
+    const proxy = ref<any>(i.props[name])
+
+    watch(
+      () => i.props[name],
+      v => (proxy.value = v)
+    )
+
+    watch(proxy, value => {
+      if (value !== i.props[name]) {
+        i.emit(`update:${name}`, value)
+      }
+    })
+
+    return proxy
+  } else {
+    return computed<any>({
```

Comment:
`computed` is largely overhead here - for perf we can use something like:

```ts
return {
    __v_isRef: true,
    get value() {
      return i.props[name]
    },
    set value(value) {
      if (value !== i.props[name]) {
        i.emit(`update:${name}`, value)
      }
    }
  } as any
```

Also the return type of `useModel` and `defineModel` can be just `Ref<T>` instead of `WritableComputedRef<T>`